### PR TITLE
Continuing on cursor work - adding a reactive decorator

### PR DESCRIPTION
**Pull request by kenotron**
- Fixes the select signature to not be generic (a bug fix)
- Adds reactive decorator for react components: both classical and stateless forms
- Adding tests to prove the functionality AND projected use inside other tests

### CODE REVIEW

**Comment by annapablo** [comment]
Reactive seems to be too generic to describe the functionality. withSelector may be?

**Comment by kenotron** [comment]
It actually wraps the observable...

File: packages/satcheljs-react/test/reactiveTests.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                expect(hello).toBe('world');
+                return <div>{foo}</div>;
+            }
+        }
+
+        let comp = new TestComponent({hello: 'world'});
+        comp.render();
+    });
+
+    it("allows classical components to be tested in a pure manner", () => {
```
Comment (kenotron) [inline packages/satcheljs-react/test/reactiveTests.tsx]: This behavior is by design - the passed in props takes precedence over the reactive decorator.

**Comment by annapablo** [comment]
Approve

**Pull request merged**